### DISCUSSION TITLE
Usage of icons in `MenuBar::label()`

### DISCUSSION THREAD

**Discussion by ghost** [NativePHP/laravel#258] [Ideas]
My app focusses mainly on displaying information in the menubar.
It would be very usefull if i could use either png, svg, or any font icon toolkit like fontawesome.
Is this possible to have this added in the future ?

**Comment by simonhamp** [comment]
To my knowledge, the menubar package that we're using in Electron only supports PNGs. It also recommends square dimensions (e.g. 20x20px) for the icon.

Can you go into more detail about your specific use-case?

**Comment by ghost** [comment]
I want to display the following in the menubar: {app_icon} {dynamic_text} {context_icon}.
I'm okay with PNG only and square dimensions.
Since my app runs from the menubar only, im using it to display useful information as well.
It's a monitoring app, and therefore a visual representation of the dynamic text would be a nice to have.

**Reply by simonhamp** [reply]
So you should be able to achieve some of this already with the [`MenuBar` class](https://github.com/NativePHP/laravel/blob/main/src/MenuBar/MenuBar.php) (which you should use via the [facade](https://github.com/NativePHP/laravel/blob/main/src/Facades/MenuBar.php)).

I don't think you can have an icon, text and _another_ icon currently, but if you wanted to hack around, the place to tinker is going to be in the NativePHP [`electron-plugin`](https://github.com/NativePHP/electron-plugin/blob/main/src/server/api/menuBar.ts).

The Electron package we're using is [`menubar`](https://github.com/max-mapper/menubar). And it all builds on top of the Electron [Tray API](https://www.electronjs.org/docs/latest/api/tray).
